### PR TITLE
Include forecasts from api database

### PR DESCRIPTION
**Pull request by dgboss**
# Test Links:
[Landing Page](https://wps-pr-2951.apps.silver.devops.gov.bc.ca/)
[MoreCast 2.0](https://wps-pr-2951.apps.silver.devops.gov.bc.ca/morecast-2)
[Percentile Calculator](https://wps-pr-2951.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-2951.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2951.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2951.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2951.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-2951.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-2951.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: api/app/tests/morecast_v2/test_utils.py
Comment (conbrad) [inline api/app/tests/morecast_v2/test_utils.py:1]: Nice tests!

File: api/app/routers/morecast_v2.py [hunk 1]
```diff
@@ -188,20 +189,33 @@ async def get_determinates_for_date_range(start_date: date,
...
 
         transformed_forecasts = transform_morecastforecastoutput_to_weatherindeterminate(
             forecasts_from_db, wfwx_stations)
-        wf1_forecasts.extend(transformed_forecasts)
+
+        # Not all weather stations report actuals at the same time, so we can end up in a situation where
+        # for a given date, we need to show the forecast from the wf1 API for one station, and the forecast
+        # from our API database for another station. We can check this by testing for the presence of an
+        # actual for the given date and station; if an actual exists we use the forecast from our API database.
+        transformed_forcasts_to_add = []
```
Comment (conbrad) [inline api/app/routers/morecast_v2.py:213]: What do you think of pulling this logic out into `app.morecast_v2.forecasts` or something and writing tests around it?

Comment (dgboss) [reply api/app/routers/morecast_v2.py]: Done: https://github.com/bcgov/wps/pull/2951/commits/8cd85436153397ae0e45595413abc7110265e053

File: api/app/morecast_v2/util.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+
+def actual_exists(transformed_forecast: WeatherIndeterminate, wf1_actuals: List[WeatherIndeterminate]):
+    # Get actual weatherdeterminates with matching station code to the forecast, then see if any of those
+    # actuals match the timestamp of the forecast
+    station_code_matches = [actual for actual in wf1_actuals if actual.station_code ==
+                            transformed_forecast.station_code]
+    utc_timestamp_matches = [station_code_match for station_code_match in station_code_matches
+                             if station_code_match.utc_timestamp == transformed_forecast.utc_timestamp]
+    if len(utc_timestamp_matches):
```
[the remark this answers is not in the document]
Comment (dgboss) [reply api/app/morecast_v2/util.py]: Good catch!
Done: https://github.com/bcgov/wps/pull/2951/commits/8cd85436153397ae0e45595413abc7110265e053

**Review by conbrad — APPROVED** [review]

**Pull request merged**